src/compiler/checker.ts

```diff
@@ -3437,7 +3442,7 @@ module ts {
             // Ternary.True if they are related with no assumptions,
             // Ternary.Maybe if they are related with assumptions of other relationships, or
             // Ternary.False if they are not related.
-            function isRelatedTo(source: Type, target: Type, reportErrors?: boolean, headMessage?: DiagnosticMessage): Ternary {
+            function isRelatedTo(source: Type, target: Type, reportErrors?: boolean, headMessage?: DiagnosticMessage, mustRecompute = false): Ternary {

```

Comment:
I think a more appropriate name is `reportAllErrors` or `elaborateErrors`. It's not that we're interested in recomputing the result (it will be the same), it's that we want all errors reported in full.